src/gui/torrentoptionsdialog.cpp

```diff
@@ -0,0 +1,444 @@
+}
+
+void TorrentOptionsDialog::accept()
+{
+    if (m_ui->radioTorrentLimit->isChecked() && !m_ui->checkMaxRatio->isChecked() && !m_ui->checkMaxTime->isChecked())
+    {
+        QMessageBox::critical(this, tr("No share limit method selected"), tr("Please select a limit method first"));
+        return;
+    }
+
+    const auto *session = BitTorrent::Session::instance();
+    for (const BitTorrent::InfoHash &hash : m_torrentHashes)
+    {
+        BitTorrent::TorrentHandle *torrent = session->findTorrent(hash);
+        if (!torrent) continue;
+
+        if (initialValues.upSpeedLimit != m_ui->spinUploadLimit->value())
+            torrent->setUploadLimit(getUploadSpeedLimit() * 1024);
+        if (initialValues.downSpeedLimit != m_ui->spinDownloadLimit->value())
+            torrent->setDownloadLimit(getDownloadSpeedLimit() * 1024);
+
+        const qreal ratioLimit = getRatio();
+        if (initialValues.ratio != ratioLimit)
+            torrent->setRatioLimit(ratioLimit);
+
+        const int seedingTimeLimit = getSeedingTime();
+        if (initialValues.seedingTime != seedingTimeLimit)
+            torrent->setSeedingTimeLimit(seedingTimeLimit);
+
+        if (!torrent->isPrivate())
+        {
+            if (initialValues.disableDHT != m_ui->checkDisableDHT->checkState())
+                torrent->setDHTDisabled(isDisableDHTChecked());
+            if (initialValues.disablePEX != m_ui->checkDisablePEX->checkState())
+                torrent->setPEXDisabled(isDisablePEXChecked());
+            if (initialValues.disableLSD != m_ui->checkDisableLSD->checkState())
+                torrent->setLSDDisabled(isDisableLSDChecked());
+        }
+    }
+
+    QDialog::accept();
+}
+
+void TorrentOptionsDialog::updateDownloadSpinValue(const int value)
+{
+    m_ui->spinDownloadLimit->setValue(value);
+}
+
+void TorrentOptionsDialog::updateUploadSpinValue(const int value)
+{
+    m_ui->spinUploadLimit->setValue(value);
+}
+
+void TorrentOptionsDialog::updateDownloadSliderValue(const int value)
+{
+    if (value > m_ui->sliderDownloadLimit->maximum())
+        m_ui->sliderDownloadLimit->setMaximum(value);
+    m_ui->sliderDownloadLimit->setValue(value);
+}
+
+void TorrentOptionsDialog::updateUploadSliderValue(const int value)
+{
+    if (value > m_ui->sliderUploadLimit->maximum())
+        m_ui->sliderUploadLimit->setMaximum(value);
+    m_ui->sliderUploadLimit->setValue(value);
+}
+
+int TorrentOptionsDialog::getDownloadSpeedLimit() const
+{
+    return qMax(0, m_ui->spinDownloadLimit->value());
+}
+
+int TorrentOptionsDialog::getUploadSpeedLimit() const
+{
+    return qMax(0, m_ui->spinUploadLimit->value());
+}
+
+bool TorrentOptionsDialog::useGlobalShareLimits() const
+{
+    return m_ui->radioUseGlobalShareLimits->isChecked();
+}
+
+qreal TorrentOptionsDialog::getRatio() const
+{
+    if (m_ui->buttonGroup->checkedId() == -1) // No radio button is selected
+        return -9.; // invalid number to signify different torrents ratios.
```

Comment:
can you drop the decimal point?

Reply:
But it's supposed to be a `qreal`. e.g.
https://github.com/qbittorrent/qBittorrent/blob/7c48ba2f1934d1f1fc280a3fcb2a98b4e46e85e7/src/base/bittorrent/torrenthandle.cpp#L45

Maybe you meant this for `getSeedingTime()`? It slipped by me there and i'll remove it.

Reply:
>But it's supposed to be a qreal. e.g.

I get it, but unless the compiler complains I would still prefer not having it (decimal point) in this case.